src/features/ProtectTransaction/__tests__/ProtectTxProtection.test.tsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import { simpleRender, fireEvent } from 'test-utils';
+
+import { translateRaw } from '@translations';
+import { USD } from '@config/fiats';
+import { bigify } from '@utils';
+
+import { ProtectTxProtectionUI, UIProps } from '../components/ProtectTxProtection';
+import { ProtectTxError } from '..';
+
+const defaultProps: UIProps = {
+  error: ProtectTxError.NO_ERROR,
+  fiat: USD,
+  isLoading: false,
+  feeAmount: { rate: 250, amount: bigify('0.002'), fee: bigify('0.002') },
+  web3Wallet: { isWeb3Wallet: true, name: 'MetaMask' },
+  isMyCryptoMember: false,
+  onCancel: jest.fn(),
+  onProtect: jest.fn()
+};
+
+const renderComponent = (props: UIProps) => {
+  return simpleRender(<ProtectTxProtectionUI {...props} />);
+};
+
+describe('ProtectTxReport', () => {
+  test('Can render default state', () => {
+    const { getByText } = renderComponent(defaultProps);
+    const btn = getByText(translateRaw('PROTECTED_TX_PROTECT_MY_TX'));
+    expect(btn).toBeInTheDocument();
+    expect(getByText(translateRaw('PROTECTED_TX_FEE'), {})).toBeInTheDocument();
+    expect(
+      getByText(
+        translateRaw('PROTECTED_TX_WEB3_WALLET_DESC', {
+          $web3WalletName: defaultProps.web3Wallet.name!
+        })
+      )
+    ).toBeInTheDocument();
+  });
+
+  test('Can render without fee', () => {
+    const { queryByText } = renderComponent({ ...defaultProps, isMyCryptoMember: true });
+    expect(queryByText(translateRaw('PROTECTED_TX_FEE'))).toBeNull();
+  });
+
+  test('Can render without web3 warning', () => {
+    const { queryByText } = renderComponent({
+      ...defaultProps,
+      web3Wallet: { isWeb3Wallet: false, name: null }
+    });
+    expect(queryByText(translateRaw('PROTECTED_TX_WEB3_WALLET_DESC'))).toBeNull();
+  });
+
+  test('Can render missing info state', () => {
+    const { getByText } = renderComponent({
+      ...defaultProps,
+      error: ProtectTxError.INSUFFICIENT_DATA
+    });
+    expect(getByText(translateRaw('MISSING_INFORMATION'))).toBeInTheDocument();
+  });
+
+  test('Can detect cancel clicks', () => {
+    const { getByText, container } = renderComponent(defaultProps);
+    fireEvent.click(container.querySelector('svg')!);
```

Comment:
I believe you can target the react component `<Icon type="cancel" />

Reply:
How?

Reply:
My bad, was confusing the syntax with SC